Issue917 system requests

JayHuLBL opened the pull request:
This closes #917. Following changes have been done:
- [x] created model to generate system requests
- [x] added two more output interfaces in model `DamperValveReheatBox`

mwetter commented on Buildings/Controls/OBC/ASHRAE/G36/Atomic/SystemRequestsReheatBox.mo:
```diff
@@ -0,0 +1,852 @@
+      iconTransformation(extent={{100,-100},{120,-80}})));
+
+  CDL.Discrete.Sampler samTCooSet(
+    final startTime=0,
```
startTime = 0 will disable the sampler if someone starts the simulation at a time smaller than zero, e.g., at startTime=86400.
We need to rewrite our sampler. I removed the startTime=0 assignment and will deal with it in https://github.com/lbl-srg/modelica-buildings/issues/938

mwetter commented on Buildings/Controls/OBC/ASHRAE/G36/Atomic/SystemRequestsReheatBox.mo:
```diff
@@ -0,0 +1,852 @@
+    "Calculate duration time when airflow setpoint is greater than zero"
+    annotation (Placement(transformation(extent={{-100,20},{-80,40}})));
+  CDL.Logical.Timer tim4 if hotWatCoi
+    "Calculate duration time when discharge air temperature is less than setpoint by 17 degC"
```
Should this be 1.7 rather than 17?

JayHuLBL replied:
It's 17 degC, as showing in G36.

mwetter commented on Buildings/Controls/OBC/ASHRAE/G36/Atomic/SystemRequestsReheatBox.mo:
```diff
@@ -0,0 +1,852 @@
+    "Check if discharge airflow setpoint is greater than 0"
+    annotation (Placement(transformation(extent={{-140,20},{-120,40}})));
+  CDL.Continuous.Hysteresis hys8(final uLow=-0.1, final uHigh=0.1) if hotWatCoi
+    "Check if discharge air temperature is 17 degC less than setpoint"
```
17 or 1.7 degC (also, better use Kelvin for differences)

mwetter commented on Buildings/Controls/OBC/ASHRAE/G36/Atomic/SystemRequestsReheatBox.mo:
```diff
@@ -0,0 +1,852 @@
+    "Calculate difference of discharge temperature (plus 8.3 degC) and its setpoint"
+    annotation (Placement(transformation(extent={{-80,-310},{-60,-290}})));
+  CDL.Continuous.AddParameter addPar(final p=17, final k=1) if hotWatCoi
+    "Discharge temperature plus 17 degC"
```
Is it really 17 Kelvin? If it is that high, we should make it a parameter

mwetter commented on Buildings/Controls/OBC/ASHRAE/G36/Atomic/Validation/SystemRequestsReheatBox.mo:
```diff
@@ -0,0 +1,115 @@
+within Buildings.Controls.OBC.ASHRAE.G36.Atomic.Validation;
+model SystemRequestsReheatBox
+  "Validate model of generating system requests"
```
This test does not validate the output for the boiler plant request. We need to include this output in the test results.

JayHuLBL commented:
@mwetter Updated the model:
- [x] changed default sample period to 3 minutes
- [x] changed the temperature difference limits to be parameters
- [x] added boiler plant request output

The validation test has no time event now, but still has 22 state events.

mwetter commented:
I can look at the state events tomorrow. Sometimes they are caused by checks of the form
`time > tNext` where `tNext` is known, but can be avoided by using `time >= tNext` as the equality allows to schedule a time event.
